bp_common/src/include/bp_common_me_if.vh

```diff
@@ -149,21 +134,34 @@ typedef enum bit [1:0]
  * lru_way_id indicates the way within the target set that will be used to fill the miss in to
  * lru_dirty indicates if the LRU way was dirty or clean when the miss request was sent
  */
-`define declare_bp_lce_cce_req_s(num_cce_mp, num_lce_mp, addr_width_mp, lce_assoc_mp, data_width_mp) \
+
+`define declare_bp_lce_cce_req_req_s(addr_width_mp, lce_assoc_mp) \
   typedef struct packed                                         \
   {                                                             \
-    logic [`BSG_SAFE_CLOG2(num_cce_mp)-1:0]      dst_id;        \
-    logic [`BSG_SAFE_CLOG2(num_lce_mp)-1:0]      src_id;        \
-    logic [data_width_mp-1:0]                    data;          \
-    bp_lce_cce_req_type_e                        msg_type;      \
+    bp_lce_cce_lru_dirty_e                       lru_dirty;     \
+    logic [`BSG_SAFE_CLOG2(lce_assoc_mp)-1:0]    lru_way_id;    \
+    logic [addr_width_mp-1:0]                    addr;          \
     bp_lce_cce_req_non_excl_e                    non_exclusive; \
+  }  bp_lce_cce_req_req_s
+
+`define declare_bp_lce_cce_req_uc_req_s(addr_width_mp, data_width_mp) \
+  typedef struct packed                                         \
+  {                                                             \
+    logic [data_width_mp-1:0]                    data;          \
+    bp_lce_cce_uc_req_size_e                     uc_size;       \
     logic [addr_width_mp-1:0]                    addr;          \
-    logic [`BSG_SAFE_CLOG2(lce_assoc_mp)-1:0]    lru_way_id;    \
-    bp_lce_cce_lru_dirty_e                       lru_dirty;     \
-    bp_lce_cce_req_non_cacheable_e               non_cacheable; \
-    bp_lce_cce_nc_req_size_e                     nc_size;       \
+  }  bp_lce_cce_req_uc_req_s
+
+`define declare_bp_lce_cce_req_s(num_cce_mp, num_lce_mp, msg_width_mp) \
+  typedef struct packed                                         \
+  {                                                             \
+    logic [msg_width_mp-1:0]                     msg;           \
```

Comment:
Can we make this a packed union?

It might be helpful to write this in the style of fe_be_if, where it's just a monolithic declare, if there's more than one parameterizable struct

Reply:
I think that requires adding padding into the structs that get union-ed, correct, because all members of the union must be the same size?

Reply:
A monolithic declare would probably be a good thing to do. Most of the structs are parameterized.

Reply:
Yeah, we would have to add padding fields to the declaration.  However, we don't have to transmit those empty bits, so it's essentially the same as we have now. The nice thing would be then we don't have to have the msg_widths as part of the struct declaration.  Just the adapters would have to know about the unpadded widths.

Maybe we can do the monolithic declare now and think more about packed unions?